lib/hexo/router.js

```diff
@@ -155,6 +155,7 @@ RouteStream.prototype._read = function() {
     }
   }).catch(function(err) {
     self.emit('error', err);
+    self.push(null);
```

Comment:
Without this line, tests hang forever in the scenario with failing renderer.

Reply:
IMO, this line is speculative, we should investigate the culprit of test hang.

Reply:
Without pushing null the stream never ends, that's the root cause of test hang. I will check how it behaves outside of test - maybe scripts is terminated in a different way, like process.exit()

Reply:
With this `push(null)` we can see the summary from the command
> INFO  0 files generated in 34 ms

Without it, the command just stops execution with unhandled rejection error printed to console.

In all other cases we close the stream, except this single case of `catch`. I think this line should be here since the beginning and it's actually a bug fix. 
What do you think @JLHwung ?

Reply:
@ertrzyiks I agree that the `stream` should be ended on any error. That is really keen observation. 👍